File: app_project/backend/app/schemas/news/announcement.py
from pydantic import BaseModel
from datetime import datetime

class AnnouncementBase(BaseModel):
    title: str
    message: str
    announcement_date: datetime

class AnnouncementCreate(AnnouncementBase):
    pass

class AnnouncementUpdate(AnnouncementBase):
    pass

class AnnouncementResponse(AnnouncementBase):
    id: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

File: app_project/frontend/src/models/news/announcement.ts
export interface AnnouncementCreate {
  title: string;
  message: string;
  announcement_date: string; // ISO date string
}

export interface AnnouncementUpdate {
  title?: string;
  message?: string;
  announcement_date?: string;
}

export interface AnnouncementResponse extends AnnouncementCreate {
  id: number;
  created_at: string;
  updated_at: string;
}
